File: experiment-14/runs/language=elixir_model=opus-4.8-fast_prompt=neutral/rep1/lib/br_soccer/competition.ex
defmodule BrSoccer.Competition do
  @moduledoc "Competition identifiers, display names and free-text parsing."

  @names %{
    brasileirao: "Brasileirão Série A",
    serie_b: "Brasileirão Série B",
    serie_c: "Brasileirão Série C",
    copa_do_brasil: "Copa do Brasil",
    libertadores: "Copa Libertadores",
    other: "Other"
  }

  @doc "Human-readable name for a competition atom."
  def name(comp), do: Map.get(@names, comp, to_string(comp))

  @doc "All known competition atoms."
  def all, do: Map.keys(@names)

  @doc """
  Parse a free-text competition name into an atom, or `nil` if unrecognised.

  Accepts the canonical atoms and a range of human spellings/aliases.
  """
  def parse(nil), do: nil
  def parse(atom) when is_atom(atom), do: if(Map.has_key?(@names, atom), do: atom, else: nil)

  def parse(text) when is_binary(text) do
    t = text |> BrSoccer.TeamName.deaccent() |> String.downcase() |> String.trim()

    cond do
      t == "" -> nil
      t in ~w(brasileirao serie_a) -> :brasileirao
      String.contains?(t, "serie a") -> :brasileirao
      String.contains?(t, "brasileir") -> :brasileirao
      String.contains?(t, "serie b") or t == "serie_b" -> :serie_b
      String.contains?(t, "serie c") or t == "serie_c" -> :serie_c
      String.contains?(t, "libertadores") -> :libertadores
      String.contains?(t, "copa do brasil") or t in ~w(cup copa) -> :copa_do_brasil
      true -> nil
    end
  end
end

File: experiments/adrianco/experiment-5/runs/language=java_model=claude-opus-4-7_tooling=none/rep1/src/main/java/com/brsoccer/mcp/model/Competition.java
package com.brsoccer.mcp.model;

public enum Competition {
    BRASILEIRAO("Brasileirão Serie A"),
    COPA_DO_BRASIL("Copa do Brasil"),
    LIBERTADORES("Copa Libertadores"),
    EXTENDED("Extended Match Statistics"),
    HISTORICAL_BRASILEIRAO("Historical Brasileirão (2003-2019)");

    private final String displayName;

    Competition(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static Competition fromString(String name) {
        if (name == null) return null;
        String n = name.toLowerCase();
        if (n.contains("libertadores")) return LIBERTADORES;
        if (n.contains("copa") || n.contains("cup")) return COPA_DO_BRASIL;
        if (n.contains("brasileirao") || n.contains("brasileirão") || n.contains("serie a")) return BRASILEIRAO;
        return null;
    }
}
